### PR TITLE
Add support for shared-file-system `/v2/services` and `/v2/scheduler-stats/pools`

### LINKED ISSUE #2349: Add support for shared-file-system `/v2/services` and `/v2/scheduler-stats/pools`
Before starting a PR, please read our [contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial).

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Fixes #2349

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://docs.openstack.org/api-ref/shared-file-system/?expanded=#list-back-end-storage-pools
https://docs.openstack.org/api-ref/shared-file-system/?expanded=#list-services

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
I tried to find the log for the failing test but I couldn't. Where can I find out why the test is failing?

**Comment by majewsky** [comment]
I can see `FAIL: TestVolumeAttachments (22.46s)` because of
```
blockstorage.go:243: Unable to delete volume 4b387b87-1a13-49be-9e9f-84eaf1fab669: Bad request with: [DELETE http://192.168.0.117/volume/v3/d4f78f9a18b84e6999283039b06b4e9c/volumes/4b387b87-1a13-49be-9e9f-84eaf1fab669], error message: {"badRequest": {"code": 400, "message": "Invalid volume: Volume status must be available or error or error_restoring or error_extending or error_managing and must not be migrating, attached, belong to a group, have snapshots or be disassociated from snapshots after volume transfer."}}
```
Also `FAIL: TestVolumes (13.58s)` because of
```
convenience.go:35: Failure in volumes_test.go, line 61: expected false but got true
```
Both of these look unrelated to the changes in this branch. @EmilienM Can you trigger a re-run, please?

**Comment by EmilienM** [comment]
@majewsky Github actions look fine to me, those are the actual jobs we care now.

I'll look at the PR asap.

**Comment by EmilienM** [comment]
I just realized we don't have acceptance coverage for this one, @majewsky any plans for doing it?

**Comment by SuperSandro2000** [comment]
We are currently upstreaming custom code in [limes](https://github.com/sapcc/limes). I have tested the endpoints within limes against our staging openstack installation. For acceptance testing I would probably need to setup my own openstack cluster for testing inside a VM which would probably take a few days. I don't think I could run the tests against our staging installation as it could interfere with other people using it.
So if possible we would like to skip adding new acceptance tests for now.

**Comment by EmilienM** [comment]
> We are currently upstreaming custom code in [limes](https://github.com/sapcc/limes). I have tested the endpoints within limes against our staging openstack installation. For acceptance testing I would probably need to setup my own openstack cluster for testing inside a VM which would probably take a few days. I don't think I could run the tests against our staging installation as it could interfere with other people using it. So if possible we would like to skip adding new acceptance tests for now.

Right now sharedfilesystems is not tested by our Github actions, I'll work on that soon. But if you have access to any OpenStack cloud with Manila you should be able to run acceptance test against it, and write new tests for this PR. If it's not possible, I'll have to check with other maintainers, but we usually have a policy of documented & tested code before it merges into master.

**Comment by SuperSandro2000** [comment]
I have pushed a commit which adds basic tests for the new things I have added in this PR. Locally I had to use an admin account to run them successful. I didn't look into the tests to see if this works but lets see what the CI run says.

**Comment by EmilienM** [comment]
@SuperSandro2000 sorry for the lag, I finally pushed a PR to enable Manila in our CI: https://github.com/gophercloud/gophercloud/pull/2361
Once this merge, this PR will need to be rebased, so we can test your code. Thanks a lot for your patience, and yes indeed our new jobs (using Github actions) are way better than what we had before, more coverage, faster, etc.

**Comment by EmilienM** [comment]
please rebase to trigger new ci jobs

**Review by mandre — COMMENTED** [review]
Hi @SuperSandro2000, thanks for the PR. It's quite a big one, and it would probably be easier in the future to split your work per API calls, so that we can merge them independently when they're ready. This looks mostly OK, I left a few comments, just one thing that surprised me was that you only implemented a subset of the supported fields, is that on purpose? I think you've already made 90% of the work and implementing the missing fields shouldn't be that much effort.

**Comment by SuperSandro2000** [comment]
> is that on purpose? I think you've already made 90% of the work and implementing the missing fields shouldn't be that much effort.

I only added the fields I needed but I can also add the other fields if you want.

File: acceptance/openstack/sharedfilesystems/v2/schedulerstats_test.go
Comment (SuperSandro2000) [inline acceptance/openstack/sharedfilesystems/v2/schedulerstats_test.go:2]:
```suggestion
//go:build acceptance || manila || schedulerstats
// +build acceptance manila schedulerstats
```

Comment (SuperSandro2000) [inline acceptance/openstack/sharedfilesystems/v2/schedulerstats_test.go:2]:
```suggestion
//go:build acceptance || manila || schedulerstats
// +build acceptance manila schedulerstats
```

**Comment by SuperSandro2000** [comment]
I am a bit clueless why the CI is failing now. Could someone assist?

File: openstack/sharedfilesystems/v2/schedulerstats/results.go
Comment (mandre) [inline openstack/sharedfilesystems/v2/schedulerstats/results.go:60]: The capabilities are only returned when querying for the pool details. I believe we should have a separate PoolDetails struct.

**Review by mandre — COMMENTED** [review]
> I am a bit clueless why the CI is failing now. Could someone assist?

You had the wrong type for several fields and that caused serialization to fail.

File: openstack/sharedfilesystems/v2/schedulerstats/results.go
Comment (SuperSandro2000) [reply openstack/sharedfilesystems/v2/schedulerstats/results.go:60]: Other details endpoint don't do it like this and only have one result. I am also not sure how that would even be done because it is always the generic paginator object returned.

**Comment by SuperSandro2000** [comment]
@mandre I finally got around fixing your comments and the tests.

**Review by mandre — APPROVED** [review]
Thanks @SuperSandro2000 for making the requested changes. This looks good and CI is happy. LGTM.

**Pull request merged**
